index.go

```diff
@@ -56,6 +56,13 @@ type CopyIndex interface {
 	// to handle necessary bookkeeping, instead of using the regular IndexReader.
 	CopyReader() CopyReader
 }
+type EventIndex interface {
+	Index
+
+	OnIndexStart()
```

Comment:
Add commentary for each API within the interface.

Reply:
refactored now and added comments